server/src/main/java/org/elasticsearch/health/HealthService.java

```diff
@@ -59,8 +67,9 @@ assert findDuplicatesByName(indicators).isEmpty()
             );
         return new HealthComponentResult(
             indicators.get(0).component(),
-            HealthStatus.merge(indicators.stream().map(HealthIndicatorResult::status)),
-            indicators
+            showComponentSummary ? HealthStatus.merge(indicators.stream().map(HealthIndicatorResult::status)) : null,
+            indicators,
+            showComponentSummary
```

Comment:
Can we avoid checking this flag again inside `HealthComponentResult` as we're already doing this here? ie. mark `HealthComponentResult#status`as @Nullable and only display it if it's non-null?
This would avoid leaking `showComponentSummary` into `HealthComponentResult`

Reply:
Yeah at one point I had more in there than just status so it was a little more complicated. I'll remove this.